### PR TITLE
wxGUI/datacatalog: Replace lambda expression with named Function to fix E731

### PR DESCRIPTION
**Pull request by arohanajit**
#### Description

This PR addresses the `E731` linting issue in `tree.py` by replacing a lambda expression with a named function. The change ensures that the code adheres to PEP 8 guidelines, which recommend using a `def` statement instead of assigning a lambda expression.

#### Changes Made

- Defined a new function `_callback` to replace the lambda expression.
- Updated the code to use the `_callback` function.

#### Rationale
- Just recreated `lambda` function with a normal `def` function as specified in the `flake8`

### CODE REVIEW

**Comment by echoix** [comment]
To review carefully. Last time I touched lambdas and gui, we ended up needing to revert.

File: gui/wxpython/datacatalog/tree.py [hunk 1]
```diff
@@ -1378,15 +1378,21 @@ def OnPasteMap(self, event):
...
+                    cMapset=self.copy_mapset[i],
+                    cMode=self.copy_mode,
+                    sMapset=self.selected_mapset[0],
+                    name=new_name,
+                ):
+                    self._onDoneReprojection(
+                        env2, gisrc2, gisrc, cLayer, cMapset, cMode, sMapset, name
+                    )
+
+                callback = _callback
```
Comment (petrasovaa) [inline gui/wxpython/datacatalog/tree.py:1395]:
```suggestion

                def callback(
                    gisrc2=gisrc2,
                    gisrc=gisrc,
                    cLayer=self.copy_layer[i],
                    cMapset=self.copy_mapset[i],
                    cMode=self.copy_mode,
                    sMapset=self.selected_mapset[0],
                    name=new_name,
                ):
                    self._onDoneReprojection(
                        env2, gisrc2, gisrc, cLayer, cMapset, cMode, sMapset, name
                    )

```

**Review by petrasovaa — COMMENTED** [review]
I think this should work, a simple example:

```python
>>> x = 1
>>> def test(x=x):
...     print(x)
... 
>>> test()
1
>>> x = 2
>>> test()
1
```

Just renaming the function (see the comment) seemed unnecessary, no?

**Review by petrasovaa — APPROVED** [review]
Looks like removing E402 is unrelated, but as long as the checks pass, that's fine I guess, probably some leftover.

**Pull request merged**